### PR TITLE
Zigbee: Build ncp sample with bootloader by default [KRKNWK-7933]

### PR DESCRIPTION
**Pull request by sebastiandraus**
Depends on:
- https://github.com/nrfconnect/sdk-nrf/pull/3822
- https://github.com/nrfconnect/sdk-nrf/pull/3832

### CODE REVIEW

**Review by wbober — CHANGES_REQUESTED** [review]
Please update SPDX tag ```LicenseRef-Nordic-5-Clause```

**Comment by sebastiandraus** [comment]
`DNM` until the https://github.com/nrfconnect/sdk-nrf/pull/3822 is merged

**Review by wbober — COMMENTED** [review]
LGTM. As discussed F2F lets enable dfu over usb when usb ncp transport is selected.

**Comment by tomchy** [comment]
#3822 merged!
Please rebase & fix conflicts.

**Review by wbober — APPROVED** [review]
LGTM

**Review by maciekfabia — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

File: samples/zigbee/ncp/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,25 @@
...
+  if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/mcuboot.conf")
+    set(mcuboot_OVERLAY_CONFIG ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/mcuboot.conf)
+  endif()
+
+  if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/app_with_mcuboot.conf")
+    set(CONF_FILE
+      prj.conf
+      ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/app_with_mcuboot.conf
+    )
+  endif()
```
Comment (tejlmand) [inline samples/zigbee/ncp/CMakeLists.txt:25]: Shall the mcuboot.conf and app_with_mcuboot.conf only be used for `usb.overlay` or the `nrf52840dongle_nrf52840` board ?

Or is this code on the wrong side of the `if ... endif` statement. 

Also, please adopt to this pattern for the child image:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_multi_image.html#image-specific-variables

and probably use a `prj_mcuboot.conf` for the main app, which allows for additional `<board>_mcuboot` Kconfig fragments to be automatically sourced:
https://docs.zephyrproject.org/latest/guides/kconfig/setting.html#the-initial-configuration

Feel free to ping me directly to discuss how to best meet your needs and keep a pattern similar to what is used elsewhere.

File: samples/zigbee/ncp/CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,25 @@
 
 cmake_minimum_required(VERSION 3.13.1)
 
+if ("usb.overlay" IN_LIST OVERLAY_CONFIG OR ${BOARD} STREQUAL "nrf52840dongle_nrf52840")
```
Comment (tejlmand) [inline samples/zigbee/ncp/CMakeLists.txt:9]: `.overlay` is for device tree files, use `overlay-usb.conf`.

**Review by tejlmand — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
Agreed that additional conf file cleanup will be addressed in followup PR.

**Pull request merged**